src/android/HealthPlugin.java

```diff
@@ -408,7 +408,11 @@ private void query(final JSONArray args, final CallbackContext callbackContext)
                     obj.put("endDate", datapoint.getEndTime(TimeUnit.MILLISECONDS));
                     if (datapoint.getDataSource() != null)
                         obj.put("source", datapoint.getDataSource().getName());
-                    DataSource hhh = datapoint.getDataSource();
+                    if (datapoint.getOriginalDataSource() != null) {
+                        DataSource dataSource = datapoint.getOriginalDataSource();
+                        String appPkgName = dataSource.getAppPackageName();
+                        obj.put("originalSourcePackage", appPkgName);
```

Comment:
I would prefer this to be `obj.put("source", appPkgName);` and delete lines 409 & 410, but `datapoint.getDataSource().getName()` might be providing useful info for other kinds of data which I haven't tested.